src/Microsoft.Extensions.Options.DataAnnotations/DataAnnotationValidateOptions.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace Microsoft.Extensions.Options
+{
+    /// <summary>
+    /// Implementation of <see cref="IValidateOptions{TOptions}"/> that uses DataAnnotation's <see cref="Validator"/> for validation.
+    /// </summary>
+    /// <typeparam name="TOptions">The instance being validated.</typeparam>
+    public class DataAnnotationValidateOptions<TOptions> : IValidateOptions<TOptions> where TOptions : class
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="name"></param>
+        public DataAnnotationValidateOptions(string name)
+        {
+            Name = name;
+        }
+
+        /// <summary>
+        /// The options name.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// Validates a specific named options instance (or all when name is null).
+        /// </summary>
+        /// <param name="name">The name of the options instance being validated.</param>
+        /// <param name="options">The options instance.</param>
+        /// <returns>The <see cref="ValidateOptionsResult"/> result.</returns>
+        public ValidateOptionsResult Validate(string name, TOptions options)
+        {
+            // Null name is used to configure all named options.
+            if (Name == null || name == Name)
+            {
+                var validationResults = new List<ValidationResult>();
+                var vc = new ValidationContext(options, serviceProvider: null, items: null);
+                var isValid = Validator.TryValidateObject(options, vc, validationResults, validateAllProperties: true);
+
+                if (isValid)
+                {
+                    return ValidateOptionsResult.Success;
+                }
+
+
+                var errors = String.Join(" : ", validationResults.Select(r => r.ErrorMessage));
```

Comment:
This is a bit icky. I'm guessing there's no "list of errors" that we can send a list of these failures to? How about put one on each line instead?

```
"Validation of the object XYZ failed:
{reason 1}
{reason 2}
{reason 3}"
```

Reply:
I think it would be also useful here to use ValidationResult.MemberNames and format that in some way so that the user knows which property(s) failed.

Reply:
Okay I'll try to clean this up, yeah the data annotation validations are treated as a single validaiton failure inside of an arbitrary amount of other validators, which is why it gets turned into a single string, since it potentially would need to be aggregated with other non data-annotation based validation errors.

Reply:
Yeah I think at least putting 1 per line would be a good separation.

Reply:
Specifically only for data annotations errors? or all of the aggregated errors as well?  I'll add a new test that demonstrates what it looks like so its easier to review in the PR

Reply:
Yeah let's see it in the test then I'll review.